tests/test__collection_api.py

```diff
@@ -2914,6 +2916,43 @@ def test__aggregate_count_errors(self):
             with self.assertRaises(mongomock.OperationFailure):
                 self.db.a.aggregate([{'$count': case}])
 
+    def test__aggregate_facet(self):
+        collection = self.db.collection
+        collection.drop()
+        collection.insert_many([
+            {
+                '_id': 1,
+                'title': 'The Pillars of Society',
+                'artist': 'Grosz',
+                'year': 1926,
+                'price': 199.99,
+            },
+            {
+                '_id': 2,
+                'title': 'Melancholy III',
+                'artist': 'Munch',
+                'year': 1902,
+                'price': 200.00,
+            },
+            {
+                '_id': 3,
+                'title': 'Melancholy III',
+                'artist': 'Munch',
+                'year': 1902,
+                'price': 200.00,
+            }
+        ])
+
+        actual = collection.aggregate([{'$group': {'_id': '$year'}},
```

Comment:
Please fix indentation to avoid ASCII art:
```py
actual = collection.aggregate([
  {'$group': ...},
  {'$facet': {
    'grouped_and_limited': ...,
    'groups_count': ...,
    ...
  },
])
```